game/theater/controlpoint.py

```diff
@@ -839,13 +840,15 @@ def capture(self, game: Game, events: GameUpdateEvents, for_player: bool) -> Non
         # All the attached TGOs have either been depopulated or captured. Tell the UI to
         # update their state. Also update orientation and IADS state for specific tgos
         for tgo in self.connected_objectives:
-            if isinstance(tgo, IadsGroundObject) or isinstance(
-                tgo, VehicleGroupGroundObject
-            ):
-                if isinstance(tgo, IadsGroundObject):
+            is_iads_go = isinstance(tgo, IadsGroundObject)
+            is_vehicle_go = isinstance(tgo, VehicleGroupGroundObject)
+            if is_iads_go or is_vehicle_go:
+                if is_iads_go:
                     game.theater.iads_network.update_tgo(tgo)
                 conflict_heading = game.theater.heading_to_conflict_from(tgo.position)
                 tgo.rotate(conflict_heading or tgo.heading)
+            elif isinstance(tgo, IadsBuildingGroundObject):
+                game.theater.iads_network.update_iads_building(tgo)
```

Comment:
This is a new part, needed to remove connections to a comms tower or power station. If I missed anything, it's probably doing the same update (as in deleting connections) whenever a comms tower or power station are destroyed, though I'm not sure if that's being taken care of already.